ansible_collections/arista/avd/docs/contribution/style-guide.md

```diff
@@ -0,0 +1,263 @@
+
+- _Example_
+
+```jinja
+{# Initial block indentation #}
+{% if my_variable is arista.avd.defined.%}
+
+{# Nested block indentation #}
+{% for ethernet_interface in ethernet_interfaces %}
+{%     if ethernet_interface.name is arista.avd.defined %}
+{%         set result = ethernet_interface.name %}
+{# ..... output truncated ..... #}
+{%     endif %}
+{% endfor %}
+```
+
+### SYNTAX-4 - Expand list on single line
+
+- _Description_
+
+Instead of doing a for loop on a single line, `join` filter should be leverage as much as possible
+
+- _Example_
+
+```jinja
+{{ ciphers | join(", ") }}
+```
+
+### SYNTAX-5 - Test if variable in a list
+
+__Description__
+
+To test if a variable is part of a list, `in` operator should be used as much as possible to avoid long `if/elif/else` block.
+
+__Example__
+
+```jinja
+{% if underlay_routing_protocol is arista.avd.defined and underlay_routing_protocol in ['ibgp', 'isis', 'ospf'] %}
+```
+
+### SYNTAX-6 - Render long CLI
+
+- _Description_
+
+When a long CLI with multiple option needs to be built, use a pure J2 logic and then print
+
+- _Example_
+
+```jinja
+{% for ip_helper in vlan_interfaces[vlan_interface].ip_helpers | arista.avd.natural_sort %}
+{%     set ip_helper_cli = "ip helper-address " ~ ip_helper %}
+{%     if vlan_interfaces[vlan_interface].ip_helpers[ip_helper].vrf is arista.avd.defined %}
+{%         set ip_helper_cli = ip_helper_cli ~ " vrf " ~ vlan_interfaces[vlan_interface].ip_helpers[ip_helper].vrf %}
+{%     endif %}
+{%     if vlan_interfaces[vlan_interface].ip_helpers[ip_helper].source_interface is arista.avd.defined %}
+{%         set ip_helper_cli = ip_helper_cli ~ " source-interface " ~ vlan_interfaces[vlan_interface].ip_helpers[ip_helper].source_interface %}
+{%     endif %}
+   {{ ip_helper_cli }}
+{% endfor %}
+```
+
+## YAML Variable definition
+
+### VAR-1 - Variable name case
+
+- _Description_
+
+All variables shall use lower case
+
+- _Example_
+
+```jinja
+{{ variable }}
+```
+
+### VAR-2 - Variable name format
+
+- _Description_
+
+If variable is multi-words, underscore `_` shall be used as a separator.
+
+- _Example_
+
+```jinja
+{{ my_variable_name }}
+```
+
+### VAR-3 - Variable for hash
+
+- _Description_
+
+For iterable variable, the plural form shall be used
+
+- _Example_
+
+```jinja
+{{ ethernet_interfaces }}
+```
+
+### VAR-4 - Variable for hash entries
+
+- _Description_
+
+For variables in a for loop, the singular form shall be used
+
+- _Example_
+
+```jinja
+{{ ethernet_interfaces[ethernet_interface] }}
+```
+
+### VAR-5 - Variables concatenation
+
+__Description__
+
+Tilde `~` symbol should be used for string concatenation as it automatically convert variables to string.
+
+__Example__
+
+```jinja
+{% set ip_helper_cli = " source-interface " ~ vlan_interfaces[vlan_interface].ip_helpers[ip_helper].source_interface %}
+```
+
+
+### VAR-6 - Variable type comparison
+
+__Description__
+
+To test type of a variable, it is recommended to use `is`/`is ne` keywords
```

Comment:
`is ne` or `is not` ? `is ne` is only used in the codebase as a string match (which is wrong according to VAR-7). `is not` is used widely.

Reply:
Fixed